python/py-cryptography/Portfile

```diff
@@ -32,18 +33,93 @@ platform macosx {
         patchfiles-append   patch-src__cffi_src_build_openssl.py.diff
     }
 }
+if {${name} ne ${subport}
+    && [string match "py*-${github.project}" ${subport}]} {
+    depends_build-append \
+                    port:py${python.version}-setuptools-rust \
+                    port:rust
+
+    depends_lib-append \
+                    path:lib/libssl.dylib:openssl \
+                    port:py${python.version}-asn1crypto \
+                    port:py${python.version}-cffi \
+                    port:py${python.version}-pip \
+                    port:py${python.version}-setuptools \
```

Comment:
why is `setuptools` not only a build dependency; and were do `py-six` and `py-pip` come from, I don't see them listedupstream, or...?

Reply:
I put `setuptools` in `depends_build`. I put `pip` in there because `port -vst install` failed without it and complained explicitly about not finding `pip`.

Reply:
sure I was suggesting to put `py-setuptools` in `depends_build` instead of `depends_lib` where you had it initially since it shouldn't be required at runtime. I still don't see where `pip` comes in, that shouldn't be needed I would think; please show the output of the build failure.

Reply:
I'm honestly not sure about that—seems like it was necessary with trace mode. I'll delete it and hope for the best.

Reply:
These trace mode and CI failures are why I believe that a pip build dependency is required:
```
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/setuptools/installer.py", line 75, in fetch_build_egg
    subprocess.check_call(cmd)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/opt/local/var/macports/build/_Users_runner_work_1_s_ports_python_py-cryptography/py39-cryptography/work/.tmp/tmpeb_zl60_', '--quiet', '--index-url', "''", '--find-links', "''", 'toml>=0.9.0']' returned non-zero exit status 1.
```